File: Reserva-Hotel-com-Front-main/ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/EstadiaHotel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ReservaDeHotel.Models
{
    public class EstadiaHotel
    {
        [Key]
        public int? IdEstadia { get; set; }
        public int? QtdQuartos { get; set; }
        public DateTime? DataEntrada { get; set; }
        public DateTime? DataSaida { get; set; }
        public List<ReservaHotel>? Reservas { get; set;}
    }
}

File: Reserva-Hotel-com-Front-main/ReservaHotelFront/src/app/EstadiaHotel.ts
import { ReservaHotel } from "./ReservaHotel";

export class EstadiaHotel {
    idEstadia: number = 0;
    qtdQuartos: number = 0;
    dataEntrada: Date = new Date();
    dataSaida: Date = new Date();
    reservaId: number = 0;
    reserva: ReservaHotel | undefined;
}
